providers/implementations/kdfs/scrypt.c

```diff
@@ -36,7 +36,7 @@ static OSSL_FUNC_kdf_get_ctx_params_fn kdf_scrypt_get_ctx_params;
 static int scrypt_alg(const char *pass, size_t passlen,
                       const unsigned char *salt, size_t saltlen,
                       uint64_t N, uint64_t r, uint64_t p, uint64_t maxmem,
-                      unsigned char *key, size_t keylen, EVP_MD *sha256,
+                      unsigned char *key, size_t keylen, const char *sha256,
```

Comment:
Why was this change needed? Is prefetching a problem?

Reply:
Instead of passing an EVP_MD, I'm passing the name.  Deep inside, it converts the EVP_MD to its name to fetch.

Reply:
Prefetching is better here.  I've reverted this big of code and reworked the bits in p5_crpt2.c